gpt_bpe_test.go

```diff
@@ -1015,6 +1052,93 @@ func TestModelDownloadPythiaSharded(t *testing.T) {
 
 }
 
+func TestModelDownloadLlama(t *testing.T) {
+	// Pythia uses a slightly different file structure, where
+	// the vocab.json and merges.txt files are stored in the
+	// tokenizer.json file. We want to check if we are able to
+	// download the model and extract the vocab.json and merges.txt
+	modelId := "georgesung/llama2_7b_chat_uncensored"
+	destPath := "./TestModelDownloadLlama"
+	destPathPTR := &destPath
+	os.RemoveAll(destPath)
+
+	var rsrcType resources.ResourceType
+	rsrcType = resources.RESOURCETYPE_TRANSFORMERS
+	hfApiToken := os.Getenv("HF_API_TOKEN")
+	os.MkdirAll(destPath, 0755)
+	_, rsrcErr := resources.ResolveResources(modelId, destPathPTR,
+		resources.RESOURCE_MODEL, rsrcType, hfApiToken)
+	if rsrcErr != nil {
+		//os.RemoveAll(destPath)
+		t.Errorf("Error downloading model resources: %s", rsrcErr)
+	}
+
+	// Check that the model files are there
+	// We want to check for the presence of the following files:
+	// config.json, pytorch_model.bin,
+	// tokenizer.json, vocab.json
+
+	// Check for config.json
+	configPath := destPath + "/config.json"
+	if _, err := os.Stat(configPath); err == nil {
+		fmt.Println("config.json exists")
+
+	} else if errors.Is(err, os.ErrNotExist) {
+		//os.RemoveAll(destPath)
+		t.Errorf("config.json does not exist")
```

Comment:
Apart from what Wes said about how this is not implemented in a brittle nor particularly extensible way, you may want to consider using a `Logger` here rather than printing to stdout.

Reply:
I'm not sure i want to rewrite the logging part of the entire test file, and the current method just calls for print statements, so I'm considering just leaving it for future me figure out if needed.